### PR TITLE
EVG-19074  Add field to distro settings to define instance profiles

### PR DESCRIPTION
**Pull request by malikchaya2**
EVG-19074

### Description
This adds a field to distro settings so that it can be passed to aws. 

### Testing
Added to existing tests 

#### Does this need documentation?
no

### CODE REVIEW

**Review by Kimchelly — CHANGES_REQUESTED** [review]
Is this supposed to be added to the distro UI?

File: model/distro/distro.go [hunk 1]
```diff
@@ -51,6 +51,7 @@ type Distro struct {
 	IsCluster             bool                  `bson:"is_cluster" json:"is_cluster" mapstructure:"is_cluster"`
 	HomeVolumeSettings    HomeVolumeSettings    `bson:"home_volume_settings" json:"home_volume_settings" mapstructure:"home_volume_settings"`
 	IceCreamSettings      IceCreamSettings      `bson:"icecream_settings,omitempty" json:"icecream_settings,omitempty" mapstructure:"icecream_settings,omitempty"`
+	IamInstanceProfileArn string                `bson:"arn,omitempty" json:"arn,omitempty" mapstructure:"arn,omitempty"`
```
[the remark this answers is not in the document]
Comment (malikchaya2) [reply model/distro/distro.go:54]: I was basing the lowercase on [this](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_IamInstanceProfileSpecification.html ). I can uppercase it.

[the remark this answers is not in the document]
Comment (malikchaya2) [reply model/distro/distro.go]: moved

File: cloud/ec2.go [hunk 2]
```diff
@@ -62,6 +62,9 @@ type EC2ProviderSettings struct {
 	// SecurityGroupIDs is a list of security group IDs.
 	SecurityGroupIDs []string `mapstructure:"security_group_ids" json:"security_group_ids,omitempty" bson:"security_group_ids,omitempty"`
 
+	// IAMInstanceProfileArn is the Amazon Resource Name (ARN) of the instance profile.
+	IAMInstanceProfileArn string `mapstructure:"iam_instance_profile_arn,omitempty" json:"iam_instance_profile_arn,omitempty"  bson:"iam_instance_profile_arn,omitempty"`
```
Comment (Kimchelly) [inline cloud/ec2.go:66]: ARN is an acronym for Amazon Resource Name so this should be `IAMInstanceProfileARN` for casing consistency.

File: apimodels/agent_models.go [hunk 3]
```diff
@@ -143,20 +143,21 @@ type CreateHost struct {
...
-	AWSKeyID        string      `mapstructure:"aws_access_key_id" json:"aws_access_key_id" yaml:"aws_access_key_id" plugin:"expand"`
-	AWSSecret       string      `mapstructure:"aws_secret_access_key" json:"aws_secret_access_key" yaml:"aws_secret_access_key" plugin:"expand"`
-	KeyName         string      `mapstructure:"key_name" json:"key_name" yaml:"key_name" plugin:"expand"`
+	AMI                   string      `mapstructure:"ami" json:"ami" yaml:"ami" plugin:"expand"`
+	Distro                string      `mapstructure:"distro" json:"distro" yaml:"distro" plugin:"expand"`
+	EBSDevices            []EbsDevice `mapstructure:"ebs_block_device" json:"ebs_block_device" yaml:"ebs_block_device" plugin:"expand"`
+	InstanceType          string      `mapstructure:"instance_type" json:"instance_type" yaml:"instance_type" plugin:"expand"`
+	IPv6                  bool        `mapstructure:"ipv6" json:"ipv6" yaml:"ipv6"`
+	Region                string      `mapstructure:"region" json:"region" yaml:"region" plugin:"expand"`
+	IAMInstanceProfileARN string      `mapstructure:"iam_instance_profile_arn" json:"iam_instance_profile_arn" yaml:"iam_instance_profile_arn" plugin:"expand"`
```
Comment (Kimchelly) [inline apimodels/agent_models.go:152]: Is this meant to be used on `host.create` hosts as well? If so, we want to add this to the `host.create` docs, as well as to Shrub.

File: rest/data/host_create.go [hunk 4]
```diff
@@ -401,6 +401,9 @@ func makeEC2IntentHost(taskID, userID, publicKey string, createHost apimodels.Cr
 	if createHost.UserdataCommand != "" {
 		ec2Settings.UserData = createHost.UserdataCommand
 	}
+	if createHost.IAMInstanceProfileARN != "" {
+		ec2Settings.IAMInstanceProfileARN = createHost.IAMInstanceProfileARN
```
Comment (Kimchelly) [inline rest/data/host_create.go:405]: Are `host.create` hosts supposed to inherit their IAM instance profile from the host running the task?

File: apimodels/agent_models.go (hunk 3 above)
Comment (malikchaya2) [reply apimodels/agent_models.go:152]: On second though, I don't think we need it for this use case. removed.

File: rest/data/host_create.go (hunk 4 above)
Comment (malikchaya2) [reply rest/data/host_create.go]: removed from host create

**Review by bynn — APPROVED** [review]
looks good!

**Review by Kimchelly — APPROVED** [review]

**Pull request merged**